src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.cs

```diff
@@ -103,7 +103,8 @@ public override void Initialize(AnalysisContext context)
                     }
 
                     // Must be literal array
-                    if (arrayCreationOperation.Initializer.ElementValues.Any(x => x is not ILiteralOperation))
+                    if (arrayCreationOperation.Initializer != null &&
```

Comment:
FYI: I am working on moving the Microsoft.CodeAnalysis package reference for NetAnalyzers project from 3.3.1 to 3.11, which should bring in the nullable annotated IOperation API surface and hence prevent such NREs in future.